File: Array/easy/check_if_array_is_sorted.cpp
//
// Created by santosh on 12/8/25.
//


#include<bits/stdc++.h>
using namespace std;

bool checkArrayIsSorted(vector<int>& nums)
{
    bool ascending = true;
    bool descending = true;
    for (int i = 0; i < nums.size() - 1; i++)
    {
        if (nums[i] > nums[i + 1])
        {
            ascending = false;
        }
        if (nums[i] < nums[i + 1])
        {
            descending = false;
        }
    }
    return ascending || descending;
}

int main()
{
    vector<int> nums = {5, 4, 6, 7, 8};
    bool ans = checkArrayIsSorted(nums);
    cout << ans << endl;
}

File: Array/easy/check_if_array_is_sorted.py
# Check if array is sorted (ascending or descending)
# Created by santosh

def check_array_is_sorted(nums: list[int]) -> bool:
    ascending = True
    descending = True
    for i in range(len(nums) - 1):
        if nums[i] > nums[i + 1]:
            ascending = False
        if nums[i] < nums[i + 1]:
            descending = False
    return ascending or descending


if __name__ == "__main__":
    nums = [5, 4, 6, 7, 8]
    ans = check_array_is_sorted(nums)
    print(ans)
